src/main/java/seedu/address/model/job/Salary.java

```diff
@@ -7,12 +7,14 @@
     public static final String MESSAGE_CONSTRAINTS = "Salary must be a string with the format "
             + "[Non-negative integers][-][Non-negative integers]\n"
             + "The right bound salary should be equal to or more than the left bound salary\n"
+            + "The maximum number of digits for the salary is 9 digits \n"
+            + "The salary cannot start with a 0 eg 01234 - 0032134\n"
             + "Salary is salary paid per month and in SGD\n"
             + "eg 4000-5000";
 
-    public static final String LEFT_REGEX = "^[ ]*[0-9]+[ ]*";
+    public static final String LEFT_REGEX = "^[ ]*[1-9][0-9]{0,8}+[ ]*";
```

Comment:
Can write some tests to test that if there are 10 numbers, it will throw an error.

Reply:
Yea just some simple changes in SalaryTest will be sufficient

Reply:
Updated test cases. Please review again.